noir_stdlib/src/hash/sha256.nr

```diff
@@ -17,82 +17,207 @@ pub fn digest<let N: u32>(msg: [u8; N]) -> [u8; 32] {
     sha256_var(msg, N as u64)
 }
 
+// Convert 64-byte array to array of 16 u32s
+fn msg_u8_to_u32(msg: [u8; 64]) -> [u32; 16] {
+    let mut msg32: [u32; 16] = [0; 16];
+
+    for i in 0..16 {
+        let mut msg_field: Field = 0;
+        for j in 0..4 {
+            msg_field = msg_field * 256 + msg[64 - 4*(i + 1) + j] as Field;
+        }
+        msg32[15 - i] = msg_field as u32;
+    }
+
+    msg32
+}
+
+unconstrained fn build_msg_block_iter<let N: u32>(
+    msg: [u8; N],
+    message_size: u64,
+    mut msg_block: [u8; 64]
+) -> ([u8; 64], u64) {
+    let mut msg_byte_ptr: u64 = 0; // Message byte pointer
+    for k in 0..N {
```

Comment:
This function looks wrong to me:
if you need several blocks, you are going to overwrite the msg_block and only get the latest one?

Reply:
Yeah we are going to rewrite it on each iteration, but I think I am doing it incorrectly here. Will push an update.

Reply:
The msg index now starts at `BLOCK_SIZE * i` where `i` is the current block we are iterating over.